### ISSUE TITLE
Ability to use an OCI Helm Chart Registry on a Rancher Catalog

### ISSUE THREAD

**Issue by Wykiki** [rancher/rancher#29105]
**What kind of request is this (question/bug/enhancement/feature request):**

Feature request

**Steps to reproduce (least amount of steps as possible):**

- Add a Rancher Catalog, providing OCI Helm Chart registry informations (https://helm.sh/docs/topics/registries/)

**Result:**

- Rancher is unable to retrieve the Charts

**Other details that may be helpful:**

We want to be able to store everything related to Kubernetes in an OCI registry.
But we encountered Rancher limitations.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.4.5
- Installation option (single install/HA): HA

**Comment by dasMulli** [comment]
This would be helpful for us as well since Azure Container Registry will retire non-OCI chart storage in the not-too-distant future.

**Comment by janeczku** [comment]
This is required to support OCI-stored helm charts from the following registries (non-conclusive list):

- Azure ACR
- AWS ECR
- Project Harbor
- Google Artifact Registry

**Comment by m4ce** [comment]
It's high time rancher supported this now..

**Comment by bluemalkin** [comment]
I would love to see this supported as I've already put all my charts in OCI and now I discover Rancher does not support it :-(

**Comment by lpmatos** [comment]
Currently my charts are in the GitLab registry. When I register it in the Rancher UI I get the message:

```bash
unsupported protocol scheme ""
```

![image](https://user-images.githubusercontent.com/58797390/107773406-6111cb00-6d1c-11eb-9421-feb1cca8c2c4.png)

**Comment by AntonioNaranjo1** [comment]
This ability would be very helpful

**Comment by m4ce** [comment]
@hitesh-agrawal

**Comment by bashofmann** [comment]
This issue is still relevant for supporting oci registries as helm chart repositories in the cluster explorer apps marketplace.

**Comment by lpmatos** [comment]
I am currently using a repository on GitLab as a repository for my helm charts, but I would like an alternative that does not involve a git repository...

**Comment by dasMulli** [comment]
> If the issue is still relevant, please add a comment to the issue so the bot can remove the label and we know it is still valid.

This issue is still relevant

**Comment by Martin-Weiss** [comment]
Just had an other customer using harbor where we store the helm charts using OCI.. so are there any plans for this feature??

**Comment by liuzhiyong1982** [comment]
we are using Azure ACR with is OCI compliant. Hopefully, Rancher can update soon. :)

**Comment by dasMulli** [comment]
This issue is still relevant

**Comment by AlexisDucastel** [comment]
Hi ! 

I do upvote this issue, OCI repositories may simplify chart management, removing the need of chartmuseum (or alternatives).

But it definitively has to be handled correctly by Rancher UI from my point of view.

Thanks.

**Comment by dasMulli** [comment]
Also note that with Helm 3.8.0 the OCI repository feature was moved out of experimental, making this feature request more important (esp. considering public registries like ACR already consider the helmv2 repo structure as deprecated and making that known on every upload)

**Comment by Ashtonian** [comment]
This issue should stay open.

**Comment by badsmoke** [comment]
did i miss something or is it still on the same stage?

**Comment by dpiraud** [comment]
Upvoting this even if we decide to move to ArgoCD or deployment purposes

**Comment by tuxpeople** [comment]
Just another longterm open Rancher issue without the people at SUSE at it, it seems. Made this experience often already. And unfortunately, once again it is something important which the people are waiting for.

Sorry for sounding like a grumpy old man. But it really looks like Rancher has some issues with the internal processes. Please guys, I know you are working hard and I'm thankfully for your work. But also try to get those things sorted out.

**Comment by mijhael3000** [comment]
This issue is still relevant

**Comment by grossdaniel85** [comment]
This is relevant also for our company.

**Comment by ScionOfDesign** [comment]
This is relevant to our company as well as we use Harbor to manage our charts. Please implement this soon.

**Comment by grossdaniel85** [comment]
Please reopen, the issue is still relevant

**Comment by grossdaniel85** [comment]
Any progress on that?

**Comment by aDisplayName** [comment]
Please, SUSE, please support OCI chart so we can use GCS / AS

**Comment by MrTinnysis** [comment]
This is relevant for us too!

**Comment by jsmonet** [comment]
This issue is still relevant, and the lack of support is making me think life might be easier forcing everything to helmfile

**Comment by todsagt** [comment]
This is very relevant to us. I have opened an issue with SUSE directly to get this moved forwards.

**Comment by ehoward** [comment]
This is relevant to us to continue to use Harbor and cloud repositories.

**Comment by ScionOfDesign** [comment]
Harbor no longer includes ChartMuseum as of v2.8.0.
This feature should be considered critical now: https://github.com/goharbor/harbor/discussions/15057

**Comment by arovetto** [comment]
Yes, please. Add the oci support. Harbor removed any other way to store helm charts

**Comment by lokidaibel** [comment]
So just updated my Harbor to 2.8x.
All my Rancher Repos are now red. 

I need OCI support. Any ETA on this ? Is there a Patch to fix this ?

**Comment by johnwc** [comment]
We ran into this as well, updated our harbor install and all chart repos are dead. When is the ETA for supporting OCI?

**Comment by rajha-korithrien** [comment]
My team has also run into this issue. Is it possible to get someone from Rancher to comment that this issue is known and being worked?

**Comment by gunamata** [comment]
Yes, This is a known issues on our backlog and there's work in progress to add support to install charts from OCI registries.

**Comment by aravol** [comment]
Running into this at our company, which uses ACR

**Comment by flostru** [comment]
bump!
Lots of charts moving to oci registries. we need this asap.
Please give us an ETA

**Comment by gunamata** [comment]
@flostru , Thanks for the bump, which continues to reaffirm our understanding of the importance of this feature. As we are in the preliminary stages of exploring approaches to add OCI support, I will not be able to provide an ETA at this time. We will pull this into an upcoming milestone as soon as we zero in on the approach. 

cc: @leflambeur @rohitsakala

**Comment by XtremeOwnageDotCom** [comment]
Bump. (Been a month since the last comment here).

Also, this issue officially turned 3 years old last week.

Happy Birthday to #29105!

**Comment by smolinari** [comment]
Leading k8s teams (like the Akuity/ Argo) team are moving to OCI. This needs real attention please. 

Scott

**Comment by geowalrus4gh** [comment]
We are evaluating a catalog system for helm installations. We are using our internal chart registry for all our chart needs. All these are OCI based. Harbor or AWS ECR is used. I thought Rancher was a natural candidate being a 
a powerful tool in k8s cluster management, but at this point since Ranchers 'Apps and Marketplace' doesn't support OCI-based helm repo, we are not considering Rancher.

**Comment by gunamata** [comment]
There's work in progress to add support for OCI charts and hopefully the first implementation would be available for preview in the next  quarter. Thank you all for your patience.

**Comment by CC007** [comment]
I assume that this is no longer targeted for 2.8.0?

**Comment by johnwc** [comment]
Is this still scheduled for this quarter?

**Comment by 0Styless** [comment]
v2.9-Next - really?

**Comment by junkiebev** [comment]
Will this get a backport?

**Comment by piyush94** [comment]
@rohitsakala  Even after the release of version 2.9.0 this issue remains.
Not able to use OCI repository at root level.
![image](https://github.com/user-attachments/assets/894a50e0-ae3a-4554-ab87-0b919a5e85de)

**Comment by smolinari** [comment]
@piyush94 - I'm using OCI repos now with now problems. Are you sure the repo is actually available?

Scott

**Comment by rajeshneo** [comment]
Unable to use ECR-based OCI charts in Rancher. Added https://github.com/rancher/rancher/issues/46984

**Comment by qaz-t** [comment]
Unable to add oci repo from harbor project level. Added #47073
